06-building-a-program-start-to-finish/writing-tests/output/invertedIndexTests.js

```diff
@@ -0,0 +1,84 @@
+const truebook = require('../example.json');
+const falseBook = require('../package.json');
+
+describe('Tests the invertedIndex.js class: ', () => {
+  const invertedIndexTrue = new InvertedIndex(truebook);
+  const invertedIndexFalse = new InvertedIndex(falseBook);
+
+  describe('Checks if file extension is json', () => {
+    it('should return true for .json file', () => {
+    expect(invertedIndexTrue.checkIfJson('example.json')).toEqual(true);
+    });
+
+    it('should return false for .txt file', () => {
+    expect(invertedIndexTrue.checkIfJson('example.txt')).toEqual(false);
+    });
+  });
+
+  describe('Checks if uploaded file contains title and text', () => {
+    it('should return true for truebook', () => {
+    expect(invertedIndexTrue.containsTitleText()).toEqual(true);
+    });
+
+    it('should return false for falsebook', () => {
+    expect(invertedIndexFalse.containsTitleText()).toEqual(false);
+    });
+  });
+
+  describe('Checks the value returned by getTitlesAndTexts()',
+  () => {
+    const returnedObject = invertedIndexTrue.getTitlesAndTexts();
+
+    it('should return true for instanceof returnedObject', () => {
+    expect(returnedObject instanceof Array).toBeTruthy();
+    });
+
+    it('should return 2', () => {
+    expect(returnedObject.length).toEqual(2);
+    });
+  });
+
+  describe('Checks if value returned by getIndexedWords() is an object', () => {
+    const returnedObject = invertedIndexTrue.getIndexedWords();
+
+    it('should return true for instanceof returnedObject', () => {
+    expect(returnedObject instanceof Array).toBeTruthy();
+    });
+  });
+
+  describe('Checks the value returned by contentToDisplay()', () => {
+    const returnedObject = invertedIndexTrue.contentToDisplay();
+    const bool = (returnedObject[0].length === returnedObject[1].length);
+
+    it('should return true if the values at indexes are equal', () => {
+    expect(bool).toEqual(true);
+    });
+
+    it('should return true for instanceof returnedObject', () => {
+    expect(returnedObject instanceof Array).toBeTruthy();
+    });
+  });
+
+  describe('Checks the value returned by displayInTableFormat()', () => {
+    const returnedObject = invertedIndexTrue.displayInTableFormat();
+    const bool = (returnedObject[0].length === returnedObject[1].length);
+
+    it('should return true if the values at indexes are equal', () => {
+    expect(bool).toEqual(true);
+    });
+
+    it('should return true for instanceof returnedObject', () => {
+    expect(returnedObject instanceof Array).toBeTruthy();
+    });
+  });
+
+  describe('Checks if the value returned by searchIndexedWords() is an Object',
+  () => {
+    const returnedObject = invertedIndexTrue.searchIndexedWords(
+      ['quick', 'An', 'unusual']);
+
+    it('should return true for instanceof returnedObject', () => {
+    expect(returnedObject instanceof Array).toBeTruthy();
+    });
+  });
+});
```

Comment:
Try to refactor your test.

Your test should use "Ubiquitous Language," which is a body of knowledge that **can be understood by both the developer and the customer**. This ubiquitous language is used to shape and develop the requirements and testing needed, **at the level of the customer's understanding**.

See language used in snippet

```
describe("Calculator: add", function (){
    it("should be able to add 2 numbers together", function (){
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", function (){
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```